core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4180,6 +4200,10 @@ int RootClingMain(int argc,
       ic++;
    }
 
+   if (sharedLibraryPathName.empty()) {
+      sharedLibraryPathName = dictpathname;
```

Comment:
Hmm, this is from the master branch, I just moved it up because I needed to replicate the TModuleGenerator logic (due to the cyclic dependency that I explained in the comment below): https://github.com/root-project/root/blob/94a28b7c2552f3428a9605649aa6d9cde22da056/core/dictgen/src/rootcling_impl.cxx#L4424